src/ALDIDigitalServices/Zed/LeanPublisher/Business/Message/MessageTransferManager.php

```diff
@@ -114,8 +129,51 @@ public function filterQueueMessageTransfers(
             }
         }
 
-        return $leanPublisherQueueMessageCollection
-            ->setValidMessages($messagesToKeep);
+        return $leanPublisherQueueMessageCollection->setValidatedMessages($messagesToKeep);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\LeanPublisherQueueMessageCollectionTransfer $leanPublisherQueueMessageCollectionTransfer
+     *
+     * @return \Generated\Shared\Transfer\LeanPublisherQueueMessageCollectionTransfer
+     */
+    public function setWriteAndDeleteMessages(
+        LeanPublisherQueueMessageCollectionTransfer $leanPublisherQueueMessageCollectionTransfer
+    ): LeanPublisherQueueMessageCollectionTransfer {
+        foreach ($leanPublisherQueueMessageCollectionTransfer->getValidatedMessages() as $message) {
+            $messageBodyTransfer = $this->getEventQueueSentMessageBodyTransfer($message);
+
+            if ($this->isDeleteEvent($messageBodyTransfer)) {
+                $leanPublisherQueueMessageCollectionTransfer->addDeleteMessage($message);
+            }
+
+            if ($this->isWriteEvent($messageBodyTransfer)) {
+                $leanPublisherQueueMessageCollectionTransfer->addWriteMessage($message);
+            }
+        }
+
+        return $leanPublisherQueueMessageCollectionTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\EventQueueSendMessageBodyTransfer $messageBodyTransfer
+     *
+     * @return bool
+     */
+    protected function isDeleteEvent(EventQueueSendMessageBodyTransfer $messageBodyTransfer): bool
+    {
+        return str_ends_with($messageBodyTransfer->getEventName(), '.delete');
```

Comment:
You could also have just made one method getMessageEvent():string that explodes by "." and returns the last entry

Reply:
You are right. Fixed.